build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java

```diff
@@ -31,12 +32,40 @@ public class GoDriver implements Serializable {
     private final Log logger;
 
     public GoDriver(String executablePath, Map<String, String> env, File workingDirectory, Log logger) {
-        this.commandExecutor = new CommandExecutor(StringUtils.defaultIfEmpty(executablePath, "go"), env);
+        this.commandExecutor = generateCommandExecutor(executablePath, env);
         this.workingDirectory = workingDirectory;
         this.logger = logger;
     }
 
+    /**
+     * Create a CommandExecutor with the given executable path and environment variables.
+     * Handle a bug in windows, where the go executable path was mistakenly considered as two command arguments
+     * in cases where the executable path contains spaces (for example: "C:\Program Files\Go\bin\go.exe").
+     *
+     * @param executablePath Go executable path
+     * @param env            Environment variables map
+     * @return CommandExecutor
+     */
+    private static CommandExecutor generateCommandExecutor(String executablePath, Map<String, String> env) {
```

Comment:
We can include tests for this function that cover all possible cases. Each test can provide an explanation of what it aims to check and why.